src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -83,73 +90,43 @@ public void close() {
         transportService.removeHandler(ACTION_NAME);
```

Comment:
this class seems to be pretty important do you think we can add a unittest for this that it really does what it should do. We can simply override the methods with assertions that do networking?